### PR TITLE
[RTM] DOC: Document workflows

### PR DESCRIPTION
**Pull request by effigies**
Beginning to address #615.

* [x] anatomical.py
* [x] base.py
* [x] bold.py (will wait on #629 to avoid excessive collision)
* [x] confounds.py
* [x] fieldmap/*.py (already done)
* [x] utils.py

### CODE REVIEW

**Comment by effigies** [comment]
Just a note: It would be good to add citations to workflow docstrings, where appropriate. If you see any that you immediately know the references for, a comment would be appreciated.

**Comment by effigies** [comment]
With the docstrings (more-or-less) in place. How do we want to organize the API documentation?

Warning: I have extremely little experience with Sphinx. Strong opinions are welcome.

**Comment by effigies** [comment]
I think all workflows are now documented. Interfaces are underway, and the rough structure of the API documentation is in place.

Also note that this exists: https://sphinx-autoapi.readthedocs.io/en/latest/

**Comment by effigies** [comment]
https://743-78558509-gh.circle-artifacts.com/1/home/ubuntu/docs/api/index.html

File: fmriprep/workflows/anatomical.py [hunk 1]
```diff
@@ -3,9 +3,25 @@
...
++++++++++++++++++++++
+
+``fmriprep`` uses FreeSurfer_ to reconstruct surfaces from T1w/T2w
+structural images.
+
+.. autofunction:: init_surface_recon_wf
+.. autofunction:: init_autorecon_resume_wf
+.. autofunction:: init_gifti_surface_wf
+
+  .. note ::
```
Comment (oesteban) [inline fmriprep/workflows/anatomical.py]: This note appears just below the gifti_surface workflow and that's misleading. Please move it to `init_anat_preproc_wf` (or maybe we should just write some acknowledgment in the About FMRIPREP and remove these notes from here and there).

File: docs/api/index.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+.. automodule:: fmriprep.workflows.base
+.. automodule:: fmriprep.workflows.anatomical
+.. automodule:: fmriprep.workflows.bold
+.. automodule:: fmriprep.workflows.util
+
+Interfaces
+----------
+
+.. autoclass:: fmriprep.interfaces.StructuralReference
```
Comment (oesteban) [inline docs/api/index.rst]: What are the plans for interfaces? Maybe in a different PR?

**Comment by effigies** [comment]
Can do interfaces in a separate pr. Put those in as examples, and one of them so that a link in gifti_surfaces_wf would resolve and not show up as an error in circle.

Can move it to its own file for now, so it resolves, but doesn't look as incomplete.

**Comment by oesteban** [comment]
That makes sense, let's leave interfaces for another PR, but keeping those two where they are right now :+1:

**Comment by effigies** [comment]
@oesteban Good to go?

**Pull request merged**